server.js

```diff
@@ -1,23 +1,81 @@
 const express = require("express");
 const cors = require("cors");
+const bp = require("body-parser");
+const fs = require("fs");
 
 const app = express();
+let data = JSON.parse(fs.readFileSync("data.json", "utf-8"));
+let maxID = Math.max(...data.map((c) => c.id));
+const messages = data;
 
 app.use(cors());
+app.use(bp.json());
+app.use(bp.urlencoded({ extended: true }));
 
-const welcomeMessage = {
-  id: 0,
-  from: "Bart",
-  text: "Welcome to CYF chat system!",
-};
-
-//This array is our "data store".
-//We will start with one message in the array.
-//Note: messages will be lost when Glitch restarts our server.
-const messages = [welcomeMessage];
+//Creates a message and post it to the database
+app.post("/messages", function (req, res) {
+  let userName = req.body.userName;
+  let userMessage = req.body.userMessage;
+  if (userName === "" || userMessage === "") {
```

Comment:
I don't think his validation will work. 

if req.body.userName or req.body.userMessage are not provided, they will not `=== ""` they will be of type `undefined`. Instead, you can make use of the fact that `""` and `undefined` are both falsey values and write:

`if (!userName || !userMessage) {`